File: python/section4/q30.py
def load_neko():
    neko = []
    sentence = []
    for line in open('../../data/neko.txt.mecab', 'r'):
        if line[:3] != 'EOS':
            surface, elems = line.split('\t')
            elems = elems.split(',')
            morpheme = {
                'surface': surface, 'base': elems[6], 'pos': elems[0].strip(), 'pos1': elems[1].strip()
            }
            sentence.append(morpheme)
        else:
            if len(sentence) != 0:
                neko.append(sentence)
            sentence = []
    return neko

if __name__ == '__main__':
    neko = load_neko()
    from pprint import pprint
    pprint(neko[:2])

File: scala/section4/src/main/scala/q30.scala
/**
  * Created by nomotoeriko on 2017/04/23.
  */
object q30 {
  def main(args: Array[String]): Unit = {
    val neko = load_neko("../../data/neko.txt.mecab")
    println(neko.take(5))
  }

  def load_neko(path: String): List[List[Map[String, String]]] = {
    val neko = scala.io.Source.fromFile(path)
    val lines = neko.getLines()
    var sentence = List.empty[Map[String, String]]
    var mor_neko = List.empty[List[Map[String, String]]]
    for(l <- lines) l match {
      case "EOS" => if(sentence != Nil) {
        mor_neko = mor_neko :+ sentence
        sentence = List.empty[Map[String, String]]
      }
      case _ => sentence = sentence :+ get_morph(l)
    }
    neko.close()
    mor_neko
  }

  def get_morph(string: String): Map[String, String] = {
    val surface = string.split("\t")(0)
    val others = string.split("\t")(1).split(",")
    Map("surface" -> surface, "base" -> others(6), "pos" -> others(0), "pos1" -> others(1))
  }
}
